### ISSUE TITLE
tensorflow-macos 2.13.0 requires typing-extensions<4.6.0,>=3.6.6, but you have typing-extensions 4.7.1 which is incompatible.

### ISSUE THREAD

**Issue by vipervs** [tensorflow/tensorflow#61848]
### Issue type

Feature Request

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.13.0

### Custom code

Yes

### OS platform and distribution

MacOS 13

### Current behavior?

I can't use pydantic because it needs typing-extensions>=4.6.1 but tensorflow-macos doesn't support typing-extensions>4.6.0

please update tensorflow-macos for newer typing-extensions

pydantic-core 2.6.3 requires typing-extensions!=4.7.0,>=4.6.0, but you have typing-extensions 4.5.0 which is incompatible.
pydantic 2.3.0 requires typing-extensions>=4.6.1, but you have typing-extensions 4.5.0 which is incompatible.

tensorflow-macos 2.13.0 requires typing-extensions<4.6.0,>=3.6.6, but you have typing-extensions 4.7.1 which is incompatible.

Thanks

### Standalone code to reproduce the issue

```shell
just install the pip packages in MacOS terminal.
```

**Comment by kenhuang1964** [comment]
Same error here! Trying to use fastapi!

**Comment by sachinprasadhs** [comment]
Hi,

Could you please create a new environment and install Tensorflow 2.13 using below commands. 
I was able to successfully install TensorFlow in 2.13 in MacOS M1.

```
python3 -m venv temp_tf
source temp_tf/bin/activate
pip install tensorflow==2.13

pip list
typing_extensions       4.5.0
```

**Comment by vipervs** [comment]
That doesn't help, we need typing-extensions>=4.6.0

**Comment by sachinprasadhs** [comment]
As the error suggests in the title, for Tensorflow MacOS 2.13,   typing-`extensions` should be <4.6.0 and >=3.6.6

**Comment by vipervs** [comment]
as described in my first post, if I use the proposed typing-extensions I cant use pydantic.

"pydantic-core 2.6.3 requires typing-extensions!=4.7.0,>=4.6.0, but you have typing-extensions 4.5.0 which is incompatible.
pydantic 2.3.0 requires typing-extensions>=4.6.1, but you have typing-extensions 4.5.0 which is incompatible."

**Comment by sachinprasadhs** [comment]
@kulinseth , Can we bump this version for tensorflow-macos.

**Comment by kenhuang1964** [comment]
I am also getting this error but I am using linux.

**Comment by avishkar7** [comment]
Hi, is there a fix for the issue yet?

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tensorflow-macos 2.13.0 requires typing-extensions<4.6.0,>=3.6.6, but you have typing-extensions 4.8.0 which is incompatible.

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pydantic-core 2.6.3 requires typing-extensions!=4.7.0,>=4.6.0, but you have typing-extensions 4.5.0 which is incompatible.
pydantic 2.3.0 requires typing-extensions>=4.6.1, but you have typing-extensions 4.5.0 which is incompatible.

**Comment by kloworizer** [comment]
same problem

**Comment by Arth-Singh** [comment]
no solution yet , even i am stucked

**Comment by dogky123** [comment]
please bump version for typing-extension

**Comment by kulinseth** [comment]
> please bump version for typing-extension

We should switch to using the official Tensorflow arm64 builds. They are available from 2.14 onwards.
cc @learning-to-play , @rishikasinha-tf , @nitins17

**Comment by sachinprasadhs** [comment]
@vipervs , Could you please install TensorFlow as per the above comment from official Tensorflow PyPi using pip install tensorflow==2.14 and close the issue. Thanks!
https://pypi.org/project/tensorflow/#files
